sockeye/inference.py

```diff
@@ -1024,11 +1038,16 @@ def translate(self, trans_inputs: List[TranslatorInput]) -> List[TranslatorOutpu
             elif len(trans_input.tokens) == 0:
                 translated_chunks.append(TranslatedChunk(id=input_idx, chunk_id=0, translation=empty_translation()))
             else:
-                # TODO(tdomhan): Remove branch without EOS with next major version bump, as future models with always be trained with source side EOS symbols
+                # TODO(tdomhan): Remove branch without EOS with next major version bump, as future models witll always be trained with source side EOS symbols
                 if self.source_with_eos:
                     max_input_length_without_eos = self.max_input_length - C.SPACE_FOR_XOS
                     # oversized input
                     if len(trans_input.tokens) > max_input_length_without_eos:
+                        if len(trans_input.tokens) > self.max_input_length:
+                            if trans_input.constraints is not None:
+                                raise RuntimeError(
+                                    "Input %d has length (%d) that exceeds max input length (%d), but it has target-side constraints, and so can't be split")
```

Comment:
So far we have avoided throwing exceptions in sockeye.translate. Is there a way to log a warning for this case and have some reasonable fallback behavior? (similar to the 'bad translator input' cases we have in the make_input functions).